### PR TITLE
Array Destruction Solution

### PR DESCRIPTION
**Pull request by SansPapyrus683**

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
ok yeah, should I add an editorial for this one?

**Comment by jessechoe10** [comment]
> ok yeah, should I add an editorial for this one?

yeah.

**Comment by SansPapyrus683** [comment]
i mean the official one seems readable enough (although it has some typos)

File: solutions/CF Array Destruction.mdx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+# Solution 
+
+**Time Complexity:** $\mathcal{O}(N^2\log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+### C++ Implementation
```
Comment (dolphingarlic) [inline solutions/CF Array Destruction.mdx:18]:
```suggestion
## C++ Implementation
```

File: solutions/CF Array Destruction.mdx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+		solve();
+	}
+}
+```
+
+</CPPSection>
+
+<JavaSection>
+
+### Java Implementation
```
Comment (dolphingarlic) [inline solutions/CF Array Destruction.mdx:94]:
```suggestion
## Java Implementation
```

File: solutions/CF Array Destruction.mdx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+				}
+			}
+
+			if (validMoves == null) {
+				System.out.println("NO");
+			} else {
+				System.out.printf("YES%n%d%n", validStart);
+				for (int[] m : validMoves) {
+					System.out.printf("%d %d%n", m[0], m[1]);
+				}
```
Comment (frodakcin) [inline solutions/CF Array Destruction.mdx:155]: yay printf :D
weird that cpp soln uses cout

(btw iirc, java printf is super slow)

**Review by frodakcin — APPROVED** [review]
All looks well (though I kinda skimmed it)

**Pull request merged**